### PR TITLE
Implement 'osnadmin channel info' subcommand and update tests #4890

### PR DESCRIPTION
**Pull request by Priyansurout**
Implemented the osnadmin channel info subcommand to retrieve detailed information about a specific channel from an Ordering Service Node (OSN). This involved modifying the CLI commands and adding corresponding functionality to interact with the channel participation API.

Type of change
New feature
Implemented the osnadmin channel info subcommand to retrieve detailed information about a specific channel from an Ordering Service Node (OSN).

Introduced info subcommand under osnadmin channel to support querying individual channel details.
Updated existing tests and added new tests to cover the new feature and edge cases.

### CODE REVIEW

File: docs/wrappers/osnadmin_channel_postscript.md [hunk 1]
```diff
@@ -51,18 +51,40 @@ Note that the system channel will always be NULL because it is no longer support
...
+  Status 200 and the details of the channels are returned.
+
+### osnadmin channel info example
+
+Here are some examples of the `osnadmin channel info` command.
+
+- Using the `--channel-id` flag to get detailed information for mychannel from the orderer at `orderer.example.com:9443`.
+
+  ```
+  osnadmin channel info -o orderer.example.com:9443 --ca-file $CA_FILE --client-cert $CLIENT_CERT --client-key $CLIENT_KEY --channel-id mychannel
```
Comment (satota2) [inline docs/wrappers/osnadmin_channel_postscript.md]:
```suggestion
  osnadmin channel info -o orderer.example.com:9443 --ca-file $CA_FILE --client-cert $CLIENT_CERT --client-key $CLIENT_KEY --channelID mychannel
```

File: docs/wrappers/osnadmin_channel_postscript.md [hunk 2]
```diff
@@ -51,18 +51,40 @@ Note that the system channel will always be NULL because it is no longer support
...
+
+  ```
+
+  Status 200 and the details of the channels are returned.
+
+### osnadmin channel info example
+
+Here are some examples of the `osnadmin channel info` command.
+
+- Using the `--channel-id` flag to get detailed information for mychannel from the orderer at `orderer.example.com:9443`.
```
Comment (satota2) [inline docs/wrappers/osnadmin_channel_postscript.md:76]:
```suggestion
- Using the `--channelID` flag to get detailed information for mychannel from the orderer at `orderer.example.com:9443`.
```

File: docs/wrappers/osnadmin_channel_postscript.md [hunk 3]
```diff
@@ -51,18 +51,40 @@ Note that the system channel will always be NULL because it is no longer support
...
+  ```
+  osnadmin channel info -o orderer.example.com:9443 --ca-file $CA_FILE --client-cert $CLIENT_CERT --client-key $CLIENT_KEY --channel-id mychannel
+
+  Status: 200
+  {
+  "name": "mychannel",
+  "url": "/participation/v1/channels/mychannel",
+  "consensusRelation": "consenter",
+  "status": "active",
+  "height": 3
```
Comment (satota2) [inline docs/wrappers/osnadmin_channel_postscript.md]:
```suggestion
	"name": "mychannel",
	"url": "/participation/v1/channels/mychannel",
	"consensusRelation": "consenter",
	"status": "active",
	"height": 3
```

File: docs/wrappers/osnadmin_channel_postscript.md [hunk 4]
```diff
@@ -51,18 +51,40 @@ Note that the system channel will always be NULL because it is no longer support
...
-	"name": "mychannel",
-	"url": "/participation/v1/channels/mychannel",
-	"consensusRelation": "consenter",
-	"status": "active",
-	"height": 3
+  "name": "mychannel",
+  "url": "/participation/v1/channels/mychannel",
+  "consensusRelation": "consenter",
+  "status": "active",
+  "height": 3
```
Comment (satota2) [inline docs/wrappers/osnadmin_channel_postscript.md:65]:
```suggestion
	"name": "mychannel",
	"url": "/participation/v1/channels/mychannel",
	"consensusRelation": "consenter",
	"status": "active",
	"height": 3
```

**Review by satota2 — CHANGES_REQUESTED** [review]
@Priyansurout Thank you for the updates.

I found some parts that needed correction, so I have added some suggestion comments in the relevant places. Please check them.

Additionally, I could not find the updates to the test code mentioned in the PR title. This PR might still be in draft status, and the updates may be added later, but I wanted to bring this to your attention just in case.

**Comment by denyeart** [comment]
All of the commits must be signed to pass DCO check. Better yet, a pull request with a single feature should ideally be delivered as a single commit. See the instructions on how to squash multiple commits into a single commit for this pull request, and how to amend commits in future pull requests - https://hyperledger-fabric.readthedocs.io/en/latest/github/github.html#updating-a-pull-request

**Comment by tock-ibm** [comment]
What is the motivation for this change? all the info about a channel can be retrieved with:

`GET  <...>/participation/v1/channels/channelID`

All that is added is another method in the cli:

`osnadmincli info -c channelID`

which is completely equivalent to the existing:

`osnadmincli list -c channelID`

as they both emit the same GET request.

So what is the point? Am I missing something here?

File: cmd/osnadmin/main.go [hunk 5]
```diff
@@ -114,10 +117,12 @@ func executeForArgs(args []string) (output string, exit int, err error) {
 		resp, err = osnadmin.Join(osnURL, marshaledConfigBlock, caCertPool, tlsClientCert)
 	case list.FullCommand():
 		if *listChannelID != "" {
-			resp, err = osnadmin.ListSingleChannel(osnURL, *listChannelID, caCertPool, tlsClientCert)
+			resp, err = osnadmin.Info(osnURL, *listChannelID, caCertPool, tlsClientCert)
```
Comment (tock-ibm) [inline cmd/osnadmin/main.go:120]: So now we have two ways of getting the same information about a channel. Why do we need that?
Renaming the existing API is not a good idea as it might break existing tools that use the current CLI.

**Comment by C0rWin** [comment]
@Priyansurout I'm following up on this pull request for the 'osnadmin channel info' subcommand.

There's been no activity on this PR since August 2024, and there are some important concerns to address. As @tock-ibm pointed out, the proposed `osnadmin info -c channelID` command appears to be functionally equivalent to the existing `osnadmin list -c channelID` command, which makes the same API call under the hood.

Without a clear rationale for adding a redundant command, it's difficult to justify merging this change. If you have specific reasons why this command would add value beyond the existing functionality, please share them.

Given the lack of activity and the unaddressed concerns, we'll consider closing this PR unless there's a response with a compelling use case that wasn't covered in the earlier discussion.

**Comment by C0rWin** [comment]
Since @tock-ibm, the desired functionality already exists, and there is no active development in this PR, I will close it, and we can always re-open it once work is revived.
